src/playlist.js

```diff
@@ -104,7 +104,7 @@ const forwardDuration = function(playlist, endSequence) {
   */
 const intervalDuration = function(playlist, endSequence, expired) {
   if (typeof endSequence === 'undefined') {
-    endSequence = playlist.mediaSequence + playlist.segments.length;
+    endSequence = playlist.mediaSequence + (playlist.segments.length - 1);
```

Comment:
endSequence is invalid if we use `playlist.segments.length` as it refers to a zero based index.